ml-agents/mlagents/trainers/torch/agent_action.py

```diff
@@ -59,3 +61,83 @@ def from_buffer(buff: AgentBuffer) -> "AgentAction":
                 discrete_tensor[..., i] for i in range(discrete_tensor.shape[-1])
             ]
         return AgentAction(continuous, discrete)
+
+    @staticmethod
+    def _padded_time_to_batch(
+        agent_buffer_field: AgentBufferField, dtype: torch.dtype = torch.float32
+    ) -> List[torch.Tensor]:
+        """
+        Pad actions and convert to tensor. Pad the data with 0s where there is no
```

Comment:
Why does this method only pad actions ? Can it be made more general ?

Reply:
After some thought - I can move it to be an AgentBufferField method, but it will make the buffer no longer `torch`-agnostic. If we're OK with this it's a decent place for it to live.

Reply:
I think I can do it with just numpy arrays.